ckanext/recombinant/datatypes.py

```diff
@@ -96,7 +96,7 @@ def canonicalize(
         # User has overridden Excel format string, probably adding currency
         # markers or digit group separators (e.g.,fr-CA uses 1$ (not $1)).
         # Accept only "DDDDD.DD", discard other characters
-        canon = re.sub(r'[$,\s]', '', str(dirty))
+        canon = re.sub(r'[$,\s]', '', str("{:.2f}".format(dirty)))
```

Comment:
@RabiaSajjad assuming that `dirty` is a float here with `:.2f`. I believe at this point, the type of dirty is always unicode. The `str` is for py3 support I believe.

I guess we could regex the decimal value for the `dirty` string?